### ISSUE TITLE
Extending meteor accounts but can't display form

### ISSUE THREAD

**Issue by maronk** [meteor-useraccounts/core#319]
This is probably something very simple, but I've been trying for a long time without success.

I have created my own login service based on the excellent tutorial here https://meteorhacks.com/extending-meteor-accounts.html

From client side
```javascript
Meteor.loginWithMysql = function(user, password, callback) {
	var loginRequest = {user: user, password: password};
	Accounts.callLoginMethod({
		methodArguments: [loginRequest],
		userCallback: callback
	});
};
```

I have then implemented the login (which is based on external accounts stored in mysql) on the server side.  This has been working perfectly in a different app in production for over 6 months.

I now want to create a new app based on the same accounts using the ionic framework.  If I add accounts-password I get the nicely laid out signup form and login form immediately.  However, I am at loss of what is the correct way to show the email and password fields for my mysql login service.  As soon as I remove accounts-password all I see is the "If you already have an account Sign in" - or "Don't have an account register" messages - but no form and no fields.

Here is my login template

```
<template name="login">

     {{#ionView}}
       {{#ionContent}}
         {{#if currentUser}}
           <div class="padding">
             <button class="button button-positive button-block" data-action="logout">Log Out</button>
           </div>
         {{else}}
           {{> atForm}}
         {{/if}}
       {{/ionContent}}
     {{/ionView}}
</template>
```

My login service works perfectly - I can login easily from the console using Meteor.login(...

I have tried too many things to list them all here - even copying the accounts-password module code (which did not work).

I must be missing something trivial here - if anybody can point me in the right direction on how to show the login form I would be grateful - this would also be a good addition to the user guide, that is how to display the login fields for custom external user services that are not oauth based.

**Comment by maronk** [comment]
Answering my own question here.  After some good morning coffee I finally figured this out.  So I'm changing this into a suggested feature.

If you look here: 
https://github.com/meteor-useraccounts/core/blob/7625af8ec508e7ebeb55424388f197487ae88574/lib/templates_helpers/at_form.js#L39

Then showPwdForm is checking for the package accounts-password.  I managed to work around this by adding this in my lib folder

```javascript
Package["accounts-password"] = {}
```

This shows the login form - I then had to create this function that refers to my other function

```javascript
Meteor.loginWithPassword = function(user, password, callback) {
  Meteor.loginWithMysql(user,password,callback);
}
```
and now login works.  

Would there be an interest in adding the option for custom login services like this on to the package.  I think in this case one would need to specify the name of the login method (in this case loginWithMysql) in configuration and also line 39 in at_form.js checking for accounts-password should look something like this based on an added configuration option called PasswordLoginImplemented (better naming anyone).

```
if (Package["accounts-password"] === undefined || AccountsTemplates.options.PasswordLoginImplemented)
```

There are many use cases for this, in our case we already have close to 100K users in our database and would not like to copy it all to mongo.

**Comment by splendido** [comment]
Hi @maronk.
I'm sorry for the late reply.

It seems you got it right, well done!

I appreciate there might be some interest in having custom login services integrated with `useraccounts`, and I think it would be a good feature to have!

My concerns are about the other calls like `resetPassword`, `changePassword` and the like...

Talking about `useraccounts@v2.0`, there will be a package called `useraccounts:password` to bring in all code and packages related to password authentication...
What about thinking about designin another package for custom login services?

**Comment by maronk** [comment]
Actually - hadn't thought about that but a generic framework around custom user accounts is a great idea.

I would assume a common interface would be to base it off accounts-password - I'll see if I can find time in the next couple of weeks to make my own code more generic and put it up on github

**Comment by splendido** [comment]
Custom login services, based off the password forms, will be implemented as plugin packages in V2.0.
The solution would be to have an `UserAccounts.password.login` method, normally pointing to `Meteor.loginWithPassword`, to be overwritten by pointing to custom services calls

**Comment by splendido** [comment]
See the [integration-with-custom-login-services](https://trello.com/c/5ar5N6Cg/13-integration-with-custom-login-services) board on trello
